### PR TITLE
Chore: Avoid stopping plugin

### LINKED ISSUE #74648: Bug: Plugin stopped just after installation
When a plugin is installed through the catalog, it's stopped just after being installed, requiring a server restart for the plugin to work. No error is printed in the log:

```
DEBUG[09-11|09:44:47] using plugin                             logger=plugin.grafana-timestream-datasource version=2
DEBUG[09-11|09:44:47] plugin address                           logger=plugin.grafana-timestream-datasource address=/tmp/plugin145898315 network=unix
DEBUG[09-11|09:44:47] Successfully started backend plugin process logger=plugin.grafana-timestream-datasource
DEBUG[09-11|09:44:47] Plugin server exited                     logger=plugin.grafana-timestream-datasource
INFO [09-11|09:44:47] plugin process exited                    logger=plugin.grafana-timestream-datasource path=/home/andres/projects/src/github.com/grafana/grafana/data/plugins/grafana-timestream-datasource/gpx_timestream_linux_amd64 pid=39274
DEBUG[09-11|09:44:47] plugin exited                            logger=plugin.grafana-timestream-datasource
```

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Plugins installed through the catalog are instantly stopped after its initialization. The reason is this change:

https://github.com/grafana/grafana/pull/73241#discussion_r1294528339

The change was: stop the plugin if in the process of starting it, the context is canceled. I have two questions:

 - Do we want to do that?
 - If we do, do you know what could be canceling the contex that is causing the plugin to stop? I couldn't find the root cause for that.

**Which issue(s) does this PR fix?**:

Fixes #74648

**Special notes for your reviewer:**

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/manager/fakes/fakes.go [hunk 1]
```diff
@@ -575,6 +578,7 @@ func (p *FakeBackendPlugin) IsManaged() bool {
 func (p *FakeBackendPlugin) Exited() bool {
 	p.mutex.RLock()
 	defer p.mutex.RUnlock()
+	go func() { p.ExitedCheckDone <- struct{}{} }()
```
Comment (wbrowne) [inline pkg/plugins/manager/fakes/fakes.go:581]: Does this need to be here too? 🤔

**Review by wbrowne — APPROVED** [review]
LGTM!

File: pkg/plugins/manager/fakes/fakes.go (hunk 1 above)
Comment (andresmgot) [reply pkg/plugins/manager/fakes/fakes.go:581]: yeah, the happy path will execute this so we ensure that the ticker has been executed but the plugin has not been stopped.

The other execution (line 555) only ensures a fast failure if the plugin is indeed stopped.

Comment (wbrowne) [reply pkg/plugins/manager/fakes/fakes.go:581]: Ah in that case may be worth adding a comment to line 555. Based on the field name it makes sense in the func, but not so much in `Stop`

File: pkg/plugins/manager/process/process.go [hunk 2]
```diff
@@ -50,40 +53,34 @@ func startPluginAndRestartKilledProcesses(ctx context.Context, p *plugins.Plugin
 	}
 
 	go func(ctx context.Context, p *plugins.Plugin) {
```
Comment (wbrowne) [inline pkg/plugins/manager/process/process.go:55]: Nit: Can remove the `ctx` arg here

File: pkg/plugins/manager/process/process_test.go [hunk 3]
```diff
@@ -71,6 +72,30 @@ func TestProcessManager_Start(t *testing.T) {
...
+	t.Run("Won't stop the plugin if the context is cancelled", func(t *testing.T) {
+		bp := fakes.NewFakeBackendPlugin(true)
+		p := createPlugin(t, bp, func(plugin *plugins.Plugin) {
+			plugin.Backend = true
+		})
+
+		keepPluginAliveTickerDuration = 1 * time.Millisecond
+		defer func() {
+			keepPluginAliveTickerDuration = 1 * time.Second
+		}()
```
Comment (wbrowne) [inline pkg/plugins/manager/process/process_test.go:85]: Nit:
```suggestion
		tickerDuration := keepPluginAliveTickerDuration
		keepPluginAliveTickerDuration = 1 * time.Millisecond
		defer func() {
			keepPluginAliveTickerDuration = tickerDuration
		}()
```

**Pull request merged**